README.md

```diff
@@ -1,4 +1,223 @@
+```
+
+If you want to use latest snapshot release from `master` branch:
+```gradle
+repositories {
+    // CoRedux snapshot repository
+    maven {
+        url "https://oss.sonatype.org/content/repositories/snapshots/"
+    }
+}
+
+implementation "com.freeletics.coredux:core:1.0.0-SNAPSHOT"
+```
+
+### Additional artifacts
+
+Following additional artifacts are also available:
+- `com.freeletics.coredux:log-android:1.0.0` - provides `LogSink` implementation, that utilizes `android.util.Log` class to print log messages
+
+## What is CoRedux
+
+CoRedux is a predictable state container, that is using same approach as [Redux](https://redux.js.org/) with
+ability to have additional side effects.
+
+Implementation is based on [Kotlin](https://kotlinlang.org/) [coroutines](https://github.com/Kotlin/kotlinx.coroutines)
+and inspired by both [RxRedux](https://github.com/freeletics/rxredux) library
+and [Coroutines in practice by Roman Elizarov](https://www.youtube.com/watch?v=a3agLJQ6vt8) KotlinConf2018 talk.
+
+### Basic concept
+
+Imagine - we need to develop a calculator app that has on new instance of app start initial value (state) `0`.
+Calculator only allows addition, deduction, multiplication and division operations.
+Operations describe what should happen with the calculator current state and can be represented as following input actions:
+```kotlin
+sealed class CalculatorAction {
+    data class Add(val value: Int) : CalculatorAction()
+    data class Deduct(val value: Int) : CalculatorAction()
+    data class Multiply(val value: Int) : CalculatorAction()
+    data class Divide(val value: Int) : CalculatorAction()
+}
+```
+
+With CoRedux you can create a store that will be a single source of current Calculator state
+and will update it on each new incoming `CalculatorAction`:
+```kotlin
+val store = coroutineScope.createStore<Int, CalculatorAction>(
+    name = "Calculator",
+    initialState = 0,
+    reducer = { currentState, newAction ->
+        when (newAction) {
+            is CalculatorAction.Add -> currentState + newAction.value
+            is CalculatorAction.Deduct -> currentState - newAction.value
+            is CalculatorAction.Multiply -> currentState * newAction.value
+            is CalculatorAction.Divide -> currentState / newAction.value
+        }
+    }
+)
+```
+
+Where `reducer` is a special function responsible for managing state. A `reducer` specify
+how the state changes in response to actions sent to the store.
+Remember that actions only describe what happened, but don't describe how the application's state changes.
+That is the job of the `reducer`.
+
+Finally you have to subscribe `StateReceiver` to the `store` instance to get the state updates over time:
+```kotlin
+store.subscribe { state -> updateUI(state) }
+```
+Where `StateReceiver` is a function, that will receive each new update of the state inside `store`.
+More then one `StateReceiver` can subscribe to the same `store` instance.
```

Comment:
> According to this code new subscribers don't get the current state of the store immediately, right?

Right, second subscriber will not receive current state on subscribe. Whenever next action will be dispatched, second subscriber will get updated state, so behaviour is similar to Rx `PublishSubject`.

> How does unsubscribing works?

Currently no unsubscribtion happens, when `CoroutineScope` is closed, but on trying to dispatch new action exception will be thrown. I think it is worth to add  a code that will remove all `StateReceiver`s from the store to prevent memory leak.